tests/devices/qutrit_mixed/test_qutrit_mixed_sampling.py

```diff
@@ -0,0 +1,561 @@
+    )
+    def test_sample_state_many_wires(self, num_wires, shuffled_wires):
+        """Tests that sample_state works as expected with many wires, and with re-ordering."""
+        shots = 10000
+        probs_shape = (QUDIT_DIM,) * num_wires
+        state_shape = probs_shape * 2
+        diagonal = np.arange(1, QUDIT_DIM**num_wires + 1, dtype=np.float64)
+        diagonal = diagonal / np.sum(diagonal)
+        state = np.diag(diagonal).reshape(state_shape)
+
+        ordered_samples = sample_state(state, shots)
+        ordered_probs = samples_to_probs(ordered_samples, num_wires)
+        assert np.allclose(ordered_probs, diagonal, atol=APPROX_ATOL)
+
+        random_samples = sample_state(state, shots, wires=shuffled_wires)
+        random_probs = samples_to_probs(random_samples, num_wires)
+
+        reordered_probs = ordered_probs.reshape(probs_shape).transpose(shuffled_wires).flatten()
+        assert np.allclose(reordered_probs, random_probs, atol=APPROX_ATOL)
+
+
+class TestMeasureWithSamples:
+    """Test that the measure_with_samples function works as expected"""
+
+    def test_sample_measure(self, two_qutrit_pure_state):
+        """Test that a sample measurement works as expected"""
+        shots = qml.measurements.Shots(100)
+        mp = qml.sample(wires=range(2))
+
+        result = measure_with_samples([mp], two_qutrit_pure_state, shots=shots)[0]
+
+        assert result.shape == (shots.total_shots, 2)
+        assert result.dtype == np.int64
+        assert_correct_sampled_two_qutrit_pure_state(result)
+
+    def test_sample_measure_single_wire(self):
+        """Test that a sample measurement on a single wire works as expected"""
+        state_vector = np.array([1, -1j, 1, 0, 0, 0, 0, 0, 0])
+        state = get_dm_of_state(state_vector, 2, 3)
+        shots = qml.measurements.Shots(100)
+
+        mp0 = qml.sample(wires=0)
+        mp1 = qml.sample(wires=1)
+
+        result0 = measure_with_samples([mp0], state, shots=shots)[0]
+        result1 = measure_with_samples([mp1], state, shots=shots)[0]
+
+        assert result0.shape == (shots.total_shots,)
+        assert result0.dtype == np.int64
+        assert np.all(result0 == 0)
+
+        assert result1.shape == (shots.total_shots,)
+        assert result1.dtype == np.int64
+        assert len(np.unique(result1)) == 3
+
+    @flaky
+    def test_approximate_sample_measure(self, two_qutrit_pure_state):
+        """Test that a sample measurement returns approximately the correct distribution"""
+        shots = qml.measurements.Shots(5000)
+        mp = qml.sample(wires=range(2))
+
+        result = measure_with_samples([mp], two_qutrit_pure_state, shots=shots, rng=123)[0]
+
+        one_or_two_prob = np.count_nonzero(result[:, 0]) / result.shape[0]
+        one_prob = np.count_nonzero(result[:, 0] == 1) / result.shape[0]
+        assert np.allclose(one_or_two_prob, 2 / 3, atol=APPROX_ATOL)
+        assert np.allclose(one_prob, 1 / 3, atol=APPROX_ATOL)
+
+    @flaky
```

Comment:
Same comment as earlier, could you discuss the rationale for number of samples / tolerance here? Does it take too long to run?

Reply:
This one could be increased and have the same APPROX_ATOL as qubits. I don't have a good reason to change it here.